File: src/main/java/com/qlik/demo/java/util/TokenUtil.java
package com.qlik.demo.java.util;

import org.springframework.security.core.context.SecurityContextHolder;

import java.util.Optional;
import java.util.UUID;

public final class TokenUtil {

    // Dot not allow instantiation
    private TokenUtil() {
    }

    public static Optional<UUID> extractUserId() {
        final var authentication = SecurityContextHolder.getContext().getAuthentication();

        if (authentication != null) {
            final var principal = authentication.getPrincipal();

            if (principal instanceof AuthUserDetails userDetails) {
                final var userId = userDetails.id();
                return Optional.ofNullable(userId)
                        .map(UUID::fromString);
            }
        }
        return Optional.empty();
    }
}

File: src/main/kotlin/com/qlik/demo/kotlin/util/TokenUtil.kt
package com.qlik.demo.kotlin.util

import org.springframework.security.core.context.SecurityContextHolder
import java.util.*


object TokenUtil {

    fun extractUserId(): UUID? {
        return SecurityContextHolder.getContext()
            .authentication
            ?.principal
            ?.let { it as? AuthUserDetails }
            ?.id
            ?.let { UUID.fromString(it) }
    }
}